mill-universal-packager/src/io/github/hoangmaihuy/mill/packager/universal/UniversalPackagerModule.scala

```diff
@@ -33,4 +34,19 @@ trait UniversalPackagerModule extends PackagerModule {
     PathRef(zip)
   }
 
+  // Create a local directory with all the files laid out as they would be in the final distribution
+  def stage: T[PathRef] = T {
+    universalMappings().foreach { case (f, p) =>
```

Comment:
This part can be refactored similar to `m2` variable in `universalPackage`, and then you can make `universalPackage` task depends on `universalStage`. By this way if we run `universalPackage` we can show `universalStage` results as well

Reply:
i think it is not so easy to reuse the `mappings` between the stage and package tasks. the `universalPackage` may have top level directory or not, but `universalStage` should always be "flat".
refactored to reuse the `universalStage`  output files in the `universalPackage`, to make sure we always have the same contents in these tasks (except the top level directory). But this doubled our IO costs.